pkg/proxy/apis/config/validation/validation.go

```diff
@@ -71,10 +71,14 @@ func Validate(config *kubeproxyconfig.KubeProxyConfiguration) field.ErrorList {
 	if config.ClusterCIDR != "" {
 		if config.FeatureGates[string(kubefeatures.IPv6DualStack)] {
 			cidrs := strings.Split(config.ClusterCIDR, ",")
-			for _, cidr := range cidrs {
-				if _, _, err := net.ParseCIDR(cidr); err != nil {
-					allErrs = append(allErrs, field.Invalid(newPath.Child("ClusterCIDR"), cidr, "must be a valid CIDR block (e.g. 10.100.0.0/16 or FD02::0:0:0/96)"))
+			if len(cidrs) == 2 {
+				for _, cidr := range cidrs {
+					if _, _, err := net.ParseCIDR(cidr); err != nil {
+						allErrs = append(allErrs, field.Invalid(newPath.Child("ClusterCIDR"), cidr, "must be a valid CIDR block (e.g. 10.100.0.0/16 or FD02::0:0:0/96)"))
```

Comment:
should we also validate that each cidr belongs to different IP family?

Reply:
I would expect so